spark-bigquery-dsv2/spark-bigquery-dsv2-common/src/main/java/com/google/cloud/spark/bigquery/v2/BaseBigQuerySource.java

```diff
@@ -15,9 +15,11 @@
  */
 package com.google.cloud.spark.bigquery.v2;
 
+import com.google.cloud.spark.bigquery.BigQueryRelationProvider;
 import org.apache.spark.sql.sources.DataSourceRegister;
 
-public abstract class BaseBigQuerySource implements DataSourceRegister {
+public abstract class BaseBigQuerySource extends BigQueryRelationProvider
```

Comment:
Please have the Spart31BigQueryTableProvider implement the interface instead of this class